### PR TITLE
Shniak - Redux and file structure

### PR DESCRIPTION
**Pull request by Warumo**
In this PR:
- File structure in my components is changed - now the test files are in a separate folder.
- Components now work and share state using Redux
- Delete and accept icons on TaskItems are now SVG with the same look in all browsers
- Minor code refactoring

### CODE REVIEW

File: src/components/Dashboard/MainContainer/MainContainer.jsx [hunk 1]
```diff
@@ -1,41 +1,27 @@
...
-    });
+    mainContainerActions.setTaskListRef(taskListRef);
+    mainContainerActions.fetchTaskList(taskListRef);
   }
 
-  handleTaskDrop = (taskListRef, taskID, newStatus) => {
-    const { taskList } = this.state;
+  handleTaskDrop = (taskID, newStatus) => {
+    const { taskList, taskListRef } = this.props;
     const taskData = taskList.filter(task => task.id === taskID)[0];
```
[the remark this answers is not in the document]
Comment (Warumo) [reply src/components/Dashboard/MainContainer/MainContainer.jsx]: That's awesome, thank you :)

File: src/actions/mainContainerActions.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { FETCH_TASKLIST, TASKLIST_REF } from './actionTypes';
+import { getTaskListAsArray } from '../components/Dashboard/MainContainer/utils';
+
+export const fetchTaskList = taskListRef => async (dispatch) => {
```
[the remark this answers is not in the document]
Comment (Warumo) [reply src/actions/mainContainerActions.js]: Yes, works perfectly without `async`. For some reason I was sure that it's still needed

File: src/components/Dashboard/TaskItem/TaskItemUtils.js [hunk 3]
```diff
@@ -1,13 +1,11 @@
 export const getTaskRef = (taskListRef, id) => {
-  const taskRef = taskListRef.child(`${id}`);
+  const taskRef = (taskListRef && Object.keys(taskListRef).length) && taskListRef.child(`${id}`);
```
Comment (Warumo) [inline src/components/Dashboard/TaskItem/TaskItemUtils.js]: Previously this part: `taskListRef && ` wasn't there, but I caught an error once on app load in this place, as for some period of time `taskListRef` is undefined (__cannot read property 'length' of undefined__). So here I have to check if the `taskListRef` is not null or undefined and then if it has some value at all. So only if those two statements are true I take a reference of the task.

File: src/components/assets/accept.svg [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="0 0 511.999 511.999" style="enable-background:new 0 0 511.999 511.999;" xml:space="preserve">
+<g>
+	<g>
+		<path d="M506.231,75.508c-7.689-7.69-20.158-7.69-27.849,0l-319.21,319.211L33.617,269.163c-7.689-7.691-20.158-7.691-27.849,0
+			c-7.69,7.69-7.69,20.158,0,27.849l139.481,139.481c7.687,7.687,20.16,7.689,27.849,0l333.133-333.136
+			C513.921,95.666,513.921,83.198,506.231,75.508z"/>
+	</g>
+</g>
+<g>
```
Comment (VitFL) [inline src/components/assets/accept.svg]: Those empty groups for elements are not necessary here, you can remove them manually, or use something like https://www.svgminify.com/ to cut all unnecessary stuff from svg.

**Pull request merged**